### PR TITLE
DT-436 Migrate Gnome-Characters to Core22/Gnome-42

### PR DESCRIPTION
**Pull request by sergio-costas**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please check only the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

**Test Configuration**:

* OS (please include version):
* Any other relevant environment information:

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: patches/patch1.diff [hunk 1]
```diff
@@ -0,0 +1,17 @@
+diff --git a/src/org.gnome.Characters.in b/src/org.gnome.Characters.in
```
Comment (hellsworth) [inline patches/patch1.diff]: Please add a comment at the top of this patch saying what it is for.

File: snapcraft.yaml [hunk 2]
```diff
@@ -18,40 +18,38 @@ slots:
...
-      snap/gnome-characters/current/usr: usr
     override-pull: |
-      snapcraftctl pull
-      snapcraftctl set-version $(git describe --tags --abbrev=10)
+      craftctl default
       sed -i -e 's|@GJS@|/usr/bin/gjs|g' src/org.gnome.Characters.in
-      sed -i -e 's|@GJS@|/usr/bin/gjs|g' src/org.gnome.Characters.BackgroundService.in
-      sed -i.bak -e 's|=@appid@$|=${SNAP}/meta/gui/org.gnome.Characters.svg|g' data/org.gnome.Characters.desktop.in
+      sed -i.bak -e 's|=@appid@$|=${SNAP}/meta/gui/org.gnome.Characters.svg|g' data/org.gnome.Characters.desktop.in.in
+      patch -p1 < $CRAFT_PROJECT_DIR/patches/patch1.diff
```
Comment (hellsworth) [inline snapcraft.yaml]: Please rename this patch to something more intuitive and use *.diff here.

**Review by hellsworth — CHANGES_REQUESTED** [review]

File: patches/patch1.diff (hunk 1 above)
Comment (sergio-costas) [reply patches/patch1.diff]: Done.

File: snapcraft.yaml (hunk 2 above)
Comment (sergio-costas) [reply snapcraft.yaml]: Done.

**Review by hellsworth — APPROVED** [review]
Looks great thanks!

**Pull request merged**